AK/BinarySearch.h

```diff
@@ -32,14 +32,16 @@
 
 namespace AK {
 
-template<typename T>
-constexpr int integral_compare(const typename RemoveConst<T>::Type& a, const typename RemoveConst<T>::Type& b)
-{
-    return a - b;
-}
+struct IntegralComperator {
```

Comment:
I think to be 100% grammatical this should be `IntegralComparator`, and `typename Comparator` below. No `e` 🙂. 

Could probably also constrain the types with some concepts, like BinaryOperator
? Did we write a set of concepts yet? 🤔 Might be a fun exercise

Reply:
more concepts? yes please!

Reply:
I've been sprinkling some here and there, but it'd be really nice to have some useful ones around at least

Reply:
I fixed the spelling mistake but I wasn't able to do the concept stuff.

I got `clang-format 11.0.0`  installed and it doesn't seem to be able to handle concepts just yet. Also, all concept code so far is surrounded with `#ifdef` macros, not sure if this is so nice.

I guess sometime in early 2021 we have to eventually drop Clang and Coverity support and can use all the nice GCC 11 features unrestricted...

Reply:
We can all dream of nice and shiny concepts in the meantime :)

Reply:
https://godbolt.org/z/ac6Thz :eyes: clang 11 seems to handle them fine though. Just clang-format giving issues?

Reply:
Here is a snippet I had trouble with:

~~~none
$ cat main.cpp
#include <concepts>

template <typename T>
concept Container = requires (T t) {
    { t.size() } -> std::same_as<std::size_t>;
};

int main() {}
$ clang-format main.cpp
#include <concepts>

template<typename T>
concept Container = requires(T t)
{
    {
        t.size()
    }
    ->std::same_as<std::size_t>;
};

int main() { }
~~~

Reply:
Looks like this patch https://reviews.llvm.org/D79773 landed on December 4th. Hopefully next release  will have it, it looks like it should fix that formatting.